core/data_center.go

```diff
@@ -16,33 +14,27 @@ type DataCenter struct {
 	Country   *Country `json:"country,omitempty"`
 }
 
-// NewDataCenterLookup takes a string that is a DataCenter ID or Permalink,
-// returning a empty *DataCenter struct with either the ID or Permalink field
-// populated with the given value. This struct is suitable as input to other
-// methods which accept a *DataCenter as input.
-func NewDataCenterLookup(
```

Comment:
Hmm, I'm almost never a fan of ambiguity (unless theres a really strong use-case argument).

I think I'd want to answer the following questions:
- How likely are we to stick with IDs prefixed with some kind of entity description?
- How much work would arise if we were to change the IDs?
- What's the size of the potential risk to users running an older version of the library, with incompatible IDs
- How much value are we actually providing by allowing ambiguity here? And what happens if a passed in string cannot be identified as one of two (or three) options?

Reply:
> - How likely are we to stick with IDs prefixed with some kind of entity description?

At this point, I'd say VERY likely. It's baked in quite deep in Katapult.

> - How much work would arise if we were to change the IDs?

Would depend on how we're changing them. If we'd be simply dropping the prefix value and keeping a random alphanumeric string, probably a lot. But if we were to switch to UUIDv4 for example, it's simple enough to verify if a string is a valid UUIDv4.

> - What's the size of the potential risk to users running an older version of the library, with incompatible IDs

Unknown, but in theory we probably wouldn't change the IDs without a new major version of the API, meaning old client code would keep working until the old API is deprecated.

> - How much value are we actually providing by allowing ambiguity here? And what happens if a passed in string cannot be identified as one of two (or three) options?

I'm leaning more and more towards it not providing much value, and in fact probably encouraging ambiguity. And I'm not sure my previous thinking of us officially providing the functionality to prevent external parties from making assumptions about our ID formats is that valid. Users who need this indeterministic form of lookup would most likely be better off not relying on a magic lookup, and instead simply trying to lookup by permalink (or relevant alt-field) first, and if that fails fall back on a lookup by ID.

TL;DR: I think we can remove the ability to turn a ambiguous lookup string into a explicitly defined lookup field/value combo.